cost-analyzer/templates/cost-analyzer-deployment-template.yaml

```diff
@@ -491,6 +491,77 @@ spec:
           - name: ubbagent-config
             mountPath: /etc/ubbagent
         {{- end }}
+        {{- if .Values.global.ammsp.enabled }}
+        - name: {{ .Values.global.ammsp.aadAuthProxy.name }}
+          image: {{ .Values.global.ammsp.aadAuthProxy.image }}
+          {{- if .Values.global.ammsp.aadAuthProxy.imagePullPolicy }}
+          imagePullPolicy: {{ .Values.global.ammsp.aadAuthProxy.imagePullPolicy }}
+          {{- else }}
+          imagePullPolicy: Always
+          {{- end }}
+          env:
+          - name: AUDIENCE
+            value: {{ .Values.global.ammsp.aadAuthProxy.audience }}
+          - name: TARGET_HOST
+            value: {{ .Values.global.ammsp.queryEndpoint }}
+          - name: LISTENING_PORT
+            value: {{ .Values.global.ammsp.aadAuthProxy.port | quote }}
+          - name: IDENTITY_TYPE
+            value: {{ .Values.global.ammsp.aadAuthProxy.identityType }}
+          {{- if eq .Values.global.ammsp.aadAuthProxy.identityType "userAssigned"}}
+          - name: AAD_CLIENT_ID
+            value: {{ required "aadClientId is required for userAssigned identity types" .Values.global.ammsp.aadAuthProxy.aadClientId | toString | trim | quote }}
+          {{- else if eq .Values.global.ammsp.aadAuthProxy.identityType "aadApplication" }}
+          - name: AAD_CLIENT_ID
+            value: {{ required "aadClientId is required for aadApplication identity types" .Values.global.ammsp.aadAuthProxy.aadClientId | toString | trim | quote }}
```

Comment:
It looks like this is a duplicate definition of `AAD_CLIENT_ID`?

Reply:
I stole this logic from [the aad-auth-proxy helm chart](https://github.com/Azure/aad-auth-proxy/blob/6c0f849997fa0d2ca3082b00710a2df130f31dc8/deploy/chart/aad-auth-proxy/templates/deployment.yaml#L48). I believe the idea is to provide `AAD_CLIENT_ID` in the `userAssigned` and `aadApplication` cases, but not otherwise. The "required" explainer bit is different in each case.

Reply:
Oh! I see that now. Thank you for the clarification.